tests/async_fixtures/test_async_fixtures_contextvars.py

```diff
@@ -0,0 +1,73 @@
+"""
+Regression test for https://github.com/pytest-dev/pytest-asyncio/issues/127:
+contextvars were not properly maintained among fixtures and tests.
+"""
+
+from __future__ import annotations
+
+import sys
+from contextlib import contextmanager
+from contextvars import ContextVar
+
+import pytest
+
+_context_var = ContextVar("context_var")
+
+
+@contextmanager
+def context_var_manager(value):
+    token = _context_var.set(value)
+    try:
+        yield
+    finally:
+        _context_var.reset(token)
+
+
+@pytest.fixture(scope="function")
```

Comment:
The test cases are well thought out! I took me some time to understand them, though. I think this is due to the fact that the test cases build upon each other.

Most of the other test files have been refactored to use _pytest.Pytester_ (refer to [this test](https://github.com/pytest-dev/pytest-asyncio/blob/c4c1b482de0472d91b40ba0c07935d704c483bce/tests/markers/test_function_scope.py) as an example). Pytester is specifically designed to test pytest plugins. Among other things, this allows isolation of tests against each other, e.g. avoiding accidentally modified context in subsequent tests.

In my opinion, the new tests should make use of Pytester. I also think the tests should be grouped according to their test intention. That will probably create some code duplication in the tests, but I think the benefits outweigh this drawback.

For example:
```python
def test_context_var_in_requesting_fixture_does_not_affect_parent_fixture(pytester: Pytester):
    pytester.makepyfile(
      dedent(
          """\
          async def no_var_fixture():
              with pytest.raises(LookupError):
                  _context_var.get()
              yield
              with pytest.raises(LookupError):
                  _context_var.get()

          @pytest_asyncio.fixture(scope="function")
          async def var_fixture_1(no_var_fixture):
              with context_var_manager("value1"):
                yield

          async def test_request_fixture(var_fixture_1):
              pass
          """
      )
    result = pytester.runpytest("--asyncio-mode=strict")  # Use strict mode, if the test isn't specific to auto mode.
    result.assert_outcomes(passed=2)
 ```

Could you give refactoring the tests a try?

Reply:
Done. There is indeed more duplication, but the purpose of each part of the test should be clearer now.